### PR TITLE
Fix FP in no-redundant-boolean

### PR DESCRIPTION
**Pull request by vilchik-elena**
Relates to #499

### CODE REVIEW

File: sonarts-core/src/rules/noRedundantBooleanRule.ts [hunk 1]
```diff
@@ -44,17 +44,23 @@ export class Rule extends tslint.Rules.TypedRule {
...
-        node.operatorToken,
-        ts.SyntaxKind.EqualsEqualsToken,
-        ts.SyntaxKind.ExclamationEqualsToken,
-        ts.SyntaxKind.BarBarToken,
-        ts.SyntaxKind.AmpersandAmpersandToken,
-      )
-    ) {
-      this.check(node.left);
-      this.check(node.right);
+    if (is(node.operatorToken, ts.SyntaxKind.BarBarToken) && node.right.getText() === "false") {
```
Comment (saberduck) [inline sonarts-core/src/rules/noRedundantBooleanRule.ts]: why not `is(node.right, ts.SyntaxKind.FalseKeyword)`;

Comment (vilchik-elena) [reply sonarts-core/src/rules/noRedundantBooleanRule.ts]: indeed

Comment (saberduck) [reply sonarts-core/src/rules/noRedundantBooleanRule.ts]: you can actually merge the ifs, and perhaps extract first condition to a small function `isOrFalse`

File: sonarts-core/src/rules/noRedundantBooleanRule.ts [hunk 2]
```diff
@@ -44,17 +44,23 @@ export class Rule extends tslint.Rules.TypedRule {
...
+      if (is(node.parent, ts.SyntaxKind.ConditionalExpression, ts.SyntaxKind.IfStatement)) {
+        this.addIssue(node.right, Rule.MESSAGE);
+      }
+    } else {
```
Comment (saberduck) [inline sonarts-core/src/rules/noRedundantBooleanRule.ts]: you can omit curly brace `{` and  write `else if` statement

**Pull request merged**